From 7795adf820964235d76dd49aa001f2ee0d8a48e2 Mon Sep 17 00:00:00 2001
From: Jason Hung <jhung@globalscaletechnologies.com>
Date: Fri, 20 Dec 2024 09:14:48 +0800
Subject: [PATCH] net: ax88179: restrict the AX88179 to support only version 0
 to 3.0

restrict the AX88179 to support only version 0 to 3.0 to prevent
ax88279 chip (ver 4.0) loading this driver.
---
 drivers/net/usb/ax88179_178a.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/usb/ax88179_178a.c b/drivers/net/usb/ax88179_178a.c
index d5ce642200e8..8ca4a51824c7 100644
--- a/drivers/net/usb/ax88179_178a.c
+++ b/drivers/net/usb/ax88179_178a.c
@@ -1870,7 +1870,7 @@ static const struct driver_info mct_info = {
 static const struct usb_device_id products[] = {
 {
 	/* ASIX AX88179 10/100/1000 */
-	USB_DEVICE(0x0b95, 0x1790),
+    USB_DEVICE_VER(0x0b95, 0x1790, 0, 0x0300),
 	.driver_info = (unsigned long)&ax88179_info,
 }, {
 	/* ASIX AX88178A 10/100/1000 */
-- 
2.34.1

